src/App.js

```diff
@@ -1,16 +1,29 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const [likesCount, setLikesCount] = useState(0);
+
+  const updateLikes = (id, like) => {
+    const messageCopy = [...messages];
+    const currentMessage = messageCopy.find((message) => message.id === id);
+    currentMessage.liked = like;
+    if (like) setLikesCount(likesCount + 1);
+    if (!like) setLikesCount(likesCount - 1);
+    setMessages(messageCopy);
+  };
```

Comment:
I like this! It avoids a second iteration through all the messages to update the likes as I am seeing in most other submissions. My initial gut feeling told me this was a bit over-engineered, but avoiding unneeded iterations is a fine reason to do a bit more engineering! So it's just engineered, I guess, lol.